apps/studio/src/server/modules/resource/resource.service.ts

```diff
@@ -597,6 +642,34 @@ export const getSearchRecentlyEdited = async ({
   })
 }
 
+export const getNestedFolderChildren = async ({
+  siteId,
+  resourceId,
+}: {
+  siteId: number
+  resourceId: number
+}): Promise<ResourceItemContent[]> => {
+  return (async function getNestedChildren(
```

Comment:
you might want to rewrite this function as a recursive CTE instead of doing it in code. This current implementation suffers from the [N+1 query problem](https://docs.sentry.io/product/issues/issue-details/performance-issues/n-one-queries/) and may cause some performance issues in the db.

also may not be consistent (since using a db connection instead of doing it in a transaction)

this [stackoverflow question](https://stackoverflow.com/questions/54907495/postgresql-recursive-parent-child-query) might have some answers (though i do also remember we already have something similar already in this repo)